File: 0268_Missing_Number/main.cpp
class Solution {
  public:
    // T = O(n), S = O(1)
    int missingNumber(vector<int> &nums) {
        int res = nums.size();
        for (int i = 0; i < nums.size(); i++) {
            res = res ^ i ^ nums[i];
        }
        return res;
    }
};

File: 0268_Missing_Number/golang/main.go
func missingNumber(nums []int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}
	res := len(nums)
	for i := 0; i < len(nums); i++ {
		res = res ^ i ^ nums[i]
	}
	return res
}